programs/server/Server.cpp

```diff
@@ -1517,13 +1518,8 @@ try
         global_context->setMMappedFileCache(mmap_cache_size);
 
     /// A cache for query results.
-    size_t query_cache_size = config().getUInt64("query_cache.size", 1_GiB);
-    if (query_cache_size)
-        global_context->setQueryCache(
-            query_cache_size,
-            config().getUInt64("query_cache.max_entries", 1024),
-            config().getUInt64("query_cache.max_entry_size", 1_MiB),
-            config().getUInt64("query_cache.max_entry_records", 30'000'000));
+    global_context->setQueryCache();
+    global_context->updateQueryCacheConfiguration(config());
```

Comment:
don't you think `setQueryCache` is loosing its meaning here? maybe make it to accept config parameter right away?